Hazel/src/Hazel/EntryPoint.h

```diff
@@ -1,6 +1,6 @@
 #pragma once
 
-#ifdef HZ_PLATFORM_WINDOWS
+#if defined HZ_PLATFORM_WINDOWS
```

Comment:
It seems a bit redundant to do platform checks when the code is the same for both platforms right now.

if there are divergences later on then they can be added at that point in time.

Reply:
> Hi Cherno,
> 
> 1. added support for Linux to the premake5.lua file
> 2. changed the "Hazel/Hazel/src/Hazel/Events/Event.h" to fix gcc compile error:
>    src/Hazel/Events/Event.h:32:83: error: pasting "::" and "WindowResize" does not give a valid preprocessing token
>    #define EVENT_CLASS_TYPE(type) static EventType GetStaticType() { return EventType::##type; }
> 3. changed files: "Hazel/src/Hazel/Core.h" and "Hazel/src/Hazel/EntryPoint.h" to add support for Linux
> 
> Thanks.
> Marian

#define EVENT_CLASS_TYPE(type) static EventType GetStaticType() { return EventType::##type; }
#define EVENT_CLASS_TYPE(type) static EventType GetStaticType() { return EventType::type; }
I am using gcc on windows i just delete '##', it works